python/across_api/auth/api.py

```diff
@@ -0,0 +1,94 @@
+# Copyright © 2023 United States Government as represented by the
+# Administrator of the National Aeronautics and Space Administration.
+# All Rights Reserved.
+
+
+import os
+from typing import Optional
+
+from jose import jwt
+from jose.exceptions import JWTError
+import httpx  # type: ignore
+from fastapi import Depends, HTTPException, Request
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+
+from ..base.api import app
+from .schema import VerifyAuth
+
+
+class JWTBearer(HTTPBearer):
+    token_endpoint: str
+    jwks_uri: str
+
+    def __init__(self, **kwargs):
+        # Configure URL from IdP
+        user_pool_id = os.environ.get("COGNITO_USER_POOL_ID")
+        if user_pool_id is not None:
+            provider_url = f"https://cognito-idp.{user_pool_id.split('_')[0]}.amazonaws.com/{user_pool_id}/"
+        elif os.environ.get("ARC_ENV") == "testing":
+            provider_url = f"http://localhost:{os.environ.get('ARC_OIDC_IDP_PORT')}/"
+        else:
+            raise RuntimeError(
+                "Environment variable COGNITO_USER_POOL_ID must be defined in production.",
+            )
+
+        # Fetch the well-known config from the IdP
+        resp = httpx.get(provider_url + ".well-known/openid-configuration")
+        resp.raise_for_status()
```

Comment:
Execute this request asynchronously. (Probably have to move it out of the constructor and into the `__call__` method.

Reply:
Fixed in 9c1510c.